src/V3String.cpp

```diff
@@ -58,19 +59,11 @@ bool VString::wildmatchi(const char* s, const char* p) VL_PURE {
 }
 
 bool VString::wildmatch(const char* s, const char* p) VL_PURE {
-    for (; *p; s++, p++) {
-        if (*p != '*') {
-            if (((*s) != (*p)) && *p != '?') return false;
-        } else {
-            // Trailing star matches everything.
-            if (!*++p) return true;
-            while (!wildmatchi(s, p)) {
-                if (*++s == '\0') return false;
-            }
-            return true;
-        }
+    if (*s == '\0') {
+        while (*p == '*') ++p;
+        return *p == '\0';
     }
-    return (*s == '\0');
+    return wildMatchImpl(s, p);
```

Comment:
I prefer the old form without an extra function call, as this function is massively hot and don't expect the compiler will inline only one template.  If you profile this refactoring standalone and it's as good, then fine.

It also seems odd to special case "**" with null source, I'd think that should fall out fine from p="***" and s="" when we've already advanced p/s and are at the end. Most strings will be non-null so likewise faster not to short circuit in first function call.

Reply:
> don't expect the compiler will inline only one template.

FWIW the template is equivalent to the original. When expanded it's 2 functions calling each other alternatingly. Don't know why the compiler would treat them differently

> If you profile this refactoring standalone and it's as good, then fine.

Please let me know if you have a test where this would show up?

> with null source, I'd think that should fall out fine from p="*" and s="" when we've already advanced p/s and are at the end.

Well, it doesn't :) `s` is always advanced before checked, so an empty string input compares whatever comes after it in memory, but I guess I can rewrite it entirely if you prefer.

Reply:
Ok, Let's jut leave it then. FWIW the old one also used to match pattern "?" true against an empty string.